pkg/3scale/amp/component/system_monitoring.go

```diff
@@ -0,0 +1,111 @@
+package component
+
+import (
+	"fmt"
+
+	"github.com/3scale/3scale-operator/pkg/assets"
+	"github.com/3scale/3scale-operator/pkg/common"
+	monitoringv1 "github.com/coreos/prometheus-operator/pkg/apis/monitoring/v1"
+	grafanav1alpha1 "github.com/integr8ly/grafana-operator/v3/pkg/apis/integreatly/v1alpha1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func (system *System) SystemSidekiqMonitoringService() *v1.Service {
+	return &v1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "system-sidekiq-metrics",
+			Labels: system.Options.SidekiqMonitoringLabels,
+		},
+		Spec: v1.ServiceSpec{
+			Ports: []v1.ServicePort{
+				v1.ServicePort{
+					Name:       "metrics",
+					Protocol:   v1.ProtocolTCP,
+					Port:       9394,
+					TargetPort: intstr.FromInt(9394),
+				},
+			},
+			Selector: map[string]string{"deploymentConfig": "system-sidekiq"},
+		},
+	}
+}
+
+func (system *System) SystemSidekiqServiceMonitor() *monitoringv1.ServiceMonitor {
+	return &monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "system-sidekiq",
+			Labels: system.Options.SidekiqMonitoringLabels,
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Endpoints: []monitoringv1.Endpoint{{
+				Port:   "metrics",
+				Path:   "/metrics",
+				Scheme: "http",
+			}},
+			Selector: metav1.LabelSelector{
+				MatchLabels: system.Options.SidekiqMonitoringLabels,
+			},
+		},
+	}
+}
+
+func SystemGrafanaDashboard(ns string) *grafanav1alpha1.GrafanaDashboard {
+	data := &struct {
+		Namespace string
+	}{
+		ns,
+	}
+	return &grafanav1alpha1.GrafanaDashboard{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "system",
+			Labels: map[string]string{
+				"monitoring-key": common.MonitoringKey,
+			},
+		},
+		Spec: grafanav1alpha1.GrafanaDashboardSpec{
+			Json: assets.TemplateAsset("monitoring/system-grafana-dashboard-1.json.tpl", data),
+			Name: fmt.Sprintf("%s/system-grafana-dashboard-1.json", ns),
+		},
+	}
+}
+
+func SystemSidekiqPrometheusRules(ns string) *monitoringv1.PrometheusRule {
+	return &monitoringv1.PrometheusRule{
+		ObjectMeta: v12.ObjectMeta{
```

Comment:
A little bit of nitpick:
```
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
```

Using `metav1` should be enough. If this happens in other files the same would apply.

Reply:
:+1: